src/main/src/main/java/org/geoserver/config/impl/ContactInfoImpl.java

```diff
@@ -152,7 +154,12 @@ public void setAddressType(String addressType) {
 
     @Override
     public String getContactEmail() {
-        return InternationalStringUtils.getOrDefault(contactEmail, internationalContactEmail);
+        String result = contactEmail;
+        if (result == null && internationalContactEmail != null)
+            result = internationalContactEmail.toString(GeoServerDefaultLocale.get());
+        String escapedResult = StringEscapeUtils.escapeHtml4(result);
```

Comment:
This should be done in the UI and in the REST freemarker templates, but not here. For example, a REST call getting the email field, returned in XML or JSON, should not have escaping happening, the field should be returned as-is in those cases.